src/htm/algorithms/SDRClassifier.cpp

```diff
@@ -165,56 +161,49 @@ void Predictor::reset() {
 }
 
 
-Predictions Predictor::infer(const UInt recordNum, const SDR &pattern)
-{
-  updateHistory_( recordNum, pattern );
-
+Predictions Predictor::infer(const SDR &pattern) const {
   Predictions result;
   for( const auto step : steps_ ) {
-    result[step] = classifiers_[step].infer( pattern );
+    result.insert({step, classifiers_.at(step).infer( pattern )});
   }
   return result;
 }
 
 
-void Predictor::learn(const UInt recordNum, const SDR &pattern,
+void Predictor::learn(const UInt recordNum, //TODO make recordNum optional, autoincrement as steps 
```

Comment:
for sequential use, even in `learn()` recordNum could be made optional

Reply:
Yea, `recordNum` is only useful if the user wants to _skip_ time steps, which is not the common case.  However when working with time series data sets, there are valid situations where you don't want to learn about a specific time-step so we should keep the time-skip capability in some form.